src/Aspire.Hosting/ApplicationModel/ContainerImageReference.cs

```diff
@@ -38,4 +39,32 @@ public ContainerImageReference(IResource resource)
     {
         return await Resource.GetFullRemoteImageNameAsync(cancellationToken).ConfigureAwait(false);
     }
+
+    /// <inheritdoc/>
+    async ValueTask<string?> IValueProvider.GetValueAsync(ValueProviderContext context, CancellationToken cancellationToken)
+    {
+        // Check if this resource is configured to save as an archive instead of pushing to a registry
+        if (context.ExecutionContext?.ServiceProvider is { } serviceProvider)
+        {
+            var logger = Resource.GetLogger(serviceProvider);
+            var buildOptionsContext = await Resource.ProcessContainerBuildOptionsCallbackAsync(
+                serviceProvider,
+                logger,
+                context.ExecutionContext,
+                cancellationToken).ConfigureAwait(false);
+
+            // For Archive destination, return the local file path
+            if (buildOptionsContext.Destination == Publishing.ContainerImageDestination.Archive)
+            {
+                if (!string.IsNullOrEmpty(buildOptionsContext.OutputPath))
+                {
+                    var imageName = buildOptionsContext.LocalImageName ?? Resource.Name.ToLowerInvariant();
+                    var imageTag = buildOptionsContext.LocalImageTag ?? "latest";
+                    return Path.Combine(buildOptionsContext.OutputPath, $"{imageName}-{imageTag}.tar");
```

Comment:
This code has a different format:

https://github.com/dotnet/aspire/blob/114d023231c7a397eb3552aa08964ec8536d4764/src/Aspire.Hosting/Publishing/DockerContainerRuntime.cs#L76

Should these be consistent?